_query-dsl/rewrite-parameter.md

```diff
@@ -0,0 +1,253 @@
+---
+layout: default
+title: Rewrite
+nav_order: 80
+---
+
+# Rewrite
+
+Multi-term queries like `wildcard`, `prefix`, `regexp`, `fuzzy`, and `range` expand internally into sets of terms. The `rewrite` parameter allows you to control how these term expansions are executed and scored.
+
+When a multi-term query expands into many terms (for example `prefix: "error*"` matching hundreds of terms), they are internally converted into term queries. This process can have the following drawbacks:
+
+* Exceed the `indices.query.bool.max_clause_count` limit (default is `1024`).
+* Affect how scores are calculated for matching documents.
+* Impact memory and latency depending on the rewrite method used.
+
+The `rewrite` parameter gives you control over how multi-term queries behave internally.
+
+| Mode                        | Scores                                 | Performance | Notes                                         |
+| --------------------------- | -------------------------------------- | ----------- | --------------------------------------------- |
+| `constant_score`            | Same score for all matches             | Best        | Default mode, ideal for filters               |
+| `scoring_boolean`           | TF/IDF-based                           | Moderate    | Full relevance scoring                        |
+| `constant_score_boolean`    | Same score, but with Boolean structure | Moderate    | Use with `must_not` or `minimum_should_match` |
+| `top_terms_N`               | TF/IDF on top N terms                  | Efficient   | Truncates expansion                           |
+| `top_terms_boost_N`         | Static boosts                          | Fast        | Less accurate                                 |
+| `top_terms_blended_freqs_N` | Blended score                          | Balanced    | Best scoring/efficiency trade-off              |
+
+
+## Available rewrite methods
+
+The following table summarizes the available rewrite methods.
+
+| Rewrite method | Description |
+| [`constant_score`](#constant_score-default) | (Default) All expanded terms are evaluated together as a single unit, assigning the same score to every match. Matching documents are not scored individually, making it very efficient for filtering use cases. |
```

Comment:
@AntonEliatra Please update the links because the headings have changed.

Reply:
@kolchfa-aws Thats update now